codice/ddf
# DDF-5162 Consolidate presentation/container React components

#### What does this PR do?
**Reviewing commit by commit may help**
Consolidates the separated container/presentation directories and keep each React component in its own directory. Notably, `react-component/presentation` still exists containing `button` and `dropdown`. These already have existing implementations in `react-component` and it's out the scope of this PR to resolve our long term approach to consolidating these components.

This PR also adds `catalog-ui-search/exports` as a sort of public Intrigue API to document which UI components are expected to be used downstream. This prevents consumers from having to depend on file locations in DDF directly to reuse components.
#### Who is reviewing it? 

@Bdthomson @samuelechu 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@djblue
@gordocanchola 
@vinamartin

#### How should this be tested?

Full UI build - start the UI and ensure that all requires are resolved correctly.
This change should be entirely deterministic - if it builds and starts we're good to go.
#### Any background context you want to provide?
The presentation/container distinction has proved useful, but having pieces of the same component located in different directories has caused confusion and difficulty finding related components. This should help clarify the distinction and make it easier to navigate to related components inside the project.
#### What are the relevant tickets?
Fixes: #5162 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
